### PR TITLE
Cloud Apps mattermost-redux feature branch

### PR DESCRIPTION
**Pull request by mickmister**
#### Summary

This PR is a base for the cloud apps feature branch. It contains:
- @larkox's [POC Plugin Integrations redux PR](https://github.com/mattermost/mattermost-redux/pull/1189)
- additions for interactive dialogs

#### Related Pull Requests

https://github.com/mattermost/mattermost-server/pull/15530
https://github.com/mattermost/mattermost-webapp/pull/6490

### CODE REVIEW

File: src/reducers/entities/integrations.ts [hunk 1]
```diff
@@ -203,10 +203,18 @@ function dialogTriggerId(state = '', action: GenericAction) {
     }
 }
 
-function dialog(state = '', action: GenericAction) {
+function dialog(state: any = null, action: GenericAction) {
```
Comment (mickmister) [inline src/reducers/entities/integrations.ts:206]: TODO: make correct type

File: src/actions/integrations.ts [hunk 2]
```diff
@@ -362,6 +362,33 @@ export function regenOAuthAppSecret(appId: string): ActionFunc {
     });
 }
 
+export function dispatchSelectInteractiveDialog(submission: DialogSubmission): ActionFunc {
```
Comment (mickmister) [inline src/actions/integrations.ts:365]: TODO: make correct type

**Review by larkox — APPROVED** [review]
LGTM! A few naming problems inherited from my POC, but your code seems fine.

**Comment by hanzei** [comment]
@mickmister and here

**Pull request closed without merging**